lib/actions/actions.js

```diff
@@ -2,29 +2,31 @@ const AddLinkToProjectAction = require('./AddLinkToProjectAction');
 const assert = require('assert');
 
 class Actions {
-  constructor(){
-    this.commandMap = {
-      'add': new AddLinkToProjectAction(),
-    }
+  constructor(commandMap){
+    this.commandMap = commandMap
   }
-
+                 
   parseActionTerm(userInputString) {
     assert(typeof userInputString === 'string');
-
     if (userInputString === ''){
-      throw new TypeError('No Term Found');
+      throw new TypeError('No Term Found')
     }
-
     let parsedUserInput = userInputString.toLowerCase().split(' ');
-
     let actionName = parsedUserInput[0]; 
-    return actionName;
+    return actionName
   }
 
   fromMessageText(actionCommand) {
     let actionName = this.parseActionTerm(actionCommand)
     return this.commandMap[actionName]
   }
+
+  static botActions(){
+    const commandMap = {
+      'add': new AddLinkToProjectAction()
+    }
+    return new Actions(commandMap)
```

Comment:
Interesting choice! What about something like this:

```
  static fromMessageText(fullMessageText) {
    const commandMap = {
      'add': new AddLinkToProjectAction()
    }

    const actionName  = Actions.parseActionTerm(fullMessageText);

    const action = commandMap[actionName];

    // validations if you like

    return action;
  }
```

Reply:
props to @yock for helped me with this approach!

by calling `const actionName  = Actions.parseActionTerm(fullMessageText);` this way, are you saying that the the parseActionTerm should be a static method as well like @yosevukilonzo mentioned.

Is there a reason why we want to be able to access the method without an instance of the class?

Reply:
you can totes do it that way and construct an instance inside `fromMessageText`. I don't want to expose the map though, based on comments below. `fromMessageText` needs to return the single `Action` instance and be the one entry point into this class.

Reply:
it's all coming together.. I'm feeling this 💯